### PR TITLE
Fix restricted Snap method implementations

### PR DESCRIPTION
**Pull request by rekmarks**
Following #132, restricted methods can no longer be bound to the requesting origin. Instead, they receive a single object parameter with an object property `context`, which in turn has a string property `origin` that identifies the the subject that made the request.

This PR ensures that all restricted method hooks use `context.origin` where necessary instead of assuming that they are bound to the requesting origin.

### CODE REVIEW

File: packages/rpc-methods/src/restricted/confirm.ts [hunk 1]
```diff
@@ -10,13 +10,14 @@ const methodName = 'snap_confirm';
 
 export type ConfirmMethodHooks = {
   /**
-   *
+   * @param snapId - The ID of the Snap that created the confirmation.
```
Comment (Gudahtt) [inline packages/rpc-methods/src/restricted/confirm.ts:13]: Is a "snapId" the same thing as an "origin"?

Comment (rekmarks) [reply packages/rpc-methods/src/restricted/confirm.ts:13]: Ah, yes. We need some terminology cleanup to happen after launch. In RPC-method-land, `origin` and `snapId` are used interchangeably. Probably we'll move to `subjectIdentifier` or something like that?

Comment (Gudahtt) [reply packages/rpc-methods/src/restricted/confirm.ts:13]: What will the snap ID be? Could it be expressed as an origin?

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: packages/rpc-methods/src/restricted/confirm.ts (hunk 1 above)
Comment (rekmarks) [reply packages/rpc-methods/src/restricted/confirm.ts:13]: We could perhaps use an origin, but in #140 I decided to encode the publishing location of the Snap in the ID itself to make it easy to do stuff like this: https://github.com/MetaMask/snaps-skunkworks/blob/snap-publishing-v0.1/packages/controllers/src/snaps/SnapController.ts/#L932-L939

**Pull request merged**